llms/googleai/vertex/vertex.go

```diff
@@ -333,76 +367,94 @@ DoStream:
 	return convertCandidates([]*genai.Candidate{candidate})
 }
 
-// convertTools converts from a list of langchaingo tools to a list of genai
-// tools.
-func convertTools(tools []llms.Tool) ([]*genai.Tool, error) {
```

Comment:
Why does this need modification? Debugging revealed no issues here, only facing the problem in https://github.com/tmc/langchaingo/pull/888. Please provide the next error message; we may have similar issues.